service/worker/scheduler/query.go

```diff
@@ -29,32 +29,44 @@ import (
 	"go.temporal.io/api/serviceerror"
 	"golang.org/x/exp/maps"
 
+	"go.temporal.io/server/common/namespace"
 	"go.temporal.io/server/common/persistence/visibility/store/elasticsearch"
```

Comment:
It feels pretty weird for code in `service/worker/scheduler` to be depending on `elasticsearch`, and for this code to actually run in frontend, not worker at all... shouldn't there be some generic function in `common/searchattribute` or `common/persistence/visibility` that can validate a query given a list of allowed non-custom search attributes, and then call that from frontend? Or just put the code in `service/frontend`?

(Yeah, I know there's GetListInfoFromStartArgs in this package called from frontend, but that feels like more of a special case since it has to share code.)

Reply:
Hm, sure. I originally put there because it's related to schedule, but I can move somewhere else. Can I do in a separate PR?